src/main/java/org/oasis_eu/portal/services/OrganizationService.java

```diff
@@ -163,6 +165,18 @@ public DCOrganization getOrganization(String dcId) {
         return null;
     }
 
+    public List<UserMembership> searchUserMembershipsFromQuery(String query){
+        List<UserMembership> organizations = new ArrayList<>();
+        String userId = userInfoService.currentUser().getUserId();
+        List<UserMembership> userMemberships = userMembershipService.getMembershipsOfUser(userId);
+        for(UserMembership userMembership : userMemberships){
```

Comment:
or a more readable :
```
        List<UserMembership> organizations = userMemberships
                .stream()
                .filter(userMembership -> userMembership.getOrganizationName().toLowerCase().contains(query.toLowerCase()))
                .collect(Collectors.toList());
```

Reply:
Just as a FYI, be careful when using `toLowerCase` et al. as this doesn't play well with all locales (most notably Turkish, where the lowercase version of **I** is **ı** and the uppercase of **i** is **İ**; and we do have Turkish users).

You should take into account the locale of the user, but this is made a bit more complicated even as it should also ideally take into account the locale of the organization name (which I'm not sure we have).

Maybe have a look at [ICU's `StringSearch`](https://ssl.icu-project.org/apiref/icu4j/com/ibm/icu/text/StringSearch.html), and possibly search once with each locale (user locale and organization locale) if we have them.

I18N is hard…